onnxruntime/test/mlas/bench/bench_util.h

```diff
@@ -5,10 +5,29 @@
 
 #include <benchmark/benchmark.h>
 
+#include <random>
+
+
 void ArgsProduct(benchmark::internal::Benchmark* bench,
                  const std::vector<std::vector<int64_t>>& arglists);
 
-std::vector<float> RandomVectorUniform(size_t N, float min_value, float max_value);
+template<typename ElementType>
+std::vector<ElementType> RandomVectorUniform(
+    size_t N,
+    ElementType min_value = std::numeric_limits<ElementType>::lowest(),
+    ElementType max_value = std::numeric_limits<ElementType>::max()) {
+  if (min_value >= max_value) {
+    return std::vector<ElementType>(N, min_value);
+  }
+  std::default_random_engine generator(static_cast<unsigned>(N));
+  std::uniform_real_distribution<double> distribution(static_cast<double>(min_value), static_cast<double>(max_value));
+  
+  std::vector<ElementType> r(N);
+  for (size_t i = 0; i < N; i++) {
+    r[i] = static_cast<ElementType>(distribution(generator));
+  }
+  return r;
+}
 
 std::vector<float> RandomVectorUniform(std::vector<int64_t> shape, float min_value, float max_value);
```

Comment:
There's already similar implementation for those random generation: [https://github.com/microsoft/onnxruntime/blob/8d737f977056444a307f1b7f0bcd402fba62d790/onnxruntime/test/common/tensor_op_test_utils.h#L28](https://github.com/microsoft/onnxruntime/blob/8d737f977056444a307f1b7f0bcd402fba62d790/onnxruntime/test/common/tensor_op_test_utils.h#L28). It'd better to leverage it

Reply:
I've tried it, had too pull too much dependencies into the projects. feel like the change is too big

Reply:
maybe we can consider moving that template into another smaller header file,